stackrox/stackrox
# ROX-19124: Added report config migration

This PR adds migration for v1 report configurations. After 4.2 acs is upgraded to 4.3, reporting 1.0 is disabled. Hence this migration is added to create v2 report config copy of v1 report configs

Manual Testing Performed
Upgrade
1.deploy 4.2 acs and create v1,v2 report configs
2.create email notifier.  follow the steps in https://github.com/charmik-redhat/reporting-scale-tests
3. upgrade to image with the changes
kubectl -n stackrox set image deploy/central central=quay.io/stackrox-io/main:4.2.x-442-ga2349cf5e2
4. verify v1 configs are copied as v2

Rollback
1. rollback 4.3 acs with migrated v1->v2 report configs to 4.2
2. verify all migrated v1 report configs are available as v2 config in tech preview

Reupgrade
1. create new v1 report configs 
2. reupgrade to 4.3
3. verify that only new v1 report configs are migrated as v2
